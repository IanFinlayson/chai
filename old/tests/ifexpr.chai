def main():
    var x = 75
    var y = "positive" if x > 0 else "negative"
    print(y)

    x = -14
    print("positive" if x > 0 else "negative")

    x = 14
    var cond = ("very" if (x if x > 0 else -x) > 100 else "somewhat") + " " + ("large" if x > 0 else "small")
    print(x, "is", cond)

    x = 150
    cond = ("very" if (x if x > 0 else -x) > 100 else "somewhat") + " " + ("large" if x > 0 else "small")
    print(x, "is", cond)

    x = -60
    cond = ("very" if (x if x > 0 else -x) > 100 else "somewhat") + " " + ("large" if x > 0 else "small")
    print(x, "is", cond)

    x = -560
    cond = ("very" if (x if x > 0 else -x) > 100 else "somewhat") + " " + ("large" if x > 0 else "small")
    print(x, "is", cond)
    
