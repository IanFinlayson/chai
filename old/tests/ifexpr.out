positive
negative
14 is somewhat large
150 is very large
-60 is somewhat small
-560 is very small
